### ISSUE TITLE
Use Coverity Scan

### ISSUE THREAD

**Issue by flx42** [BVLC/caffe#2590]
Hello all,

To improve the robustness of the code and catch more bugs, I suggest we use [Coverity Scan](http://www.coverity.com/products/coverity-scan/), it's free for open source projects and can be integrated with Github and Travis.
Since I know the team is busy this week, I went ahead and created the project in Coverity Scan. If the maintainers don't want to use this tool, I will delete the project; if you are interested I will add you to the list of owners of the project.

**Comment by lukeyeager** [comment]
Somewhat relatedly, we could use [Coveralls](https://coveralls.io/features) to check the test coverage. It might motivate more people to provide tests with their pull requests.

**Comment by longjon** [comment]
Cool, thanks @flx42! Indeed I don't know when we'll have time to put into integration, but we're fans of static analysis, so feel free to add us, make PRs, or take whatever other steps you feel would be useful, and we'll see how it goes.

@lukeyeager, yes, this is #1177. We also welcome PRs for this.

**Comment by flx42** [comment]
I made the Caffe project public on Coverity Scan:
https://scan.coverity.com/projects/5405

@longjon: I added you as an owner of the project.

The project kinda looks weird right now, when submitting a GPU build, it says 11M lines of code were analyzed (maybe because of the compiled CUDA code?). I submitted builds with/without GPU enabled so it's probably why the project history looks weird.

I used the following command to generate the build to submit:
```
cov-build --dir cov-int make -j 12 distribute test
```
In `Makefile.config`, I set `DEBUG := 1`.

Next steps would be to take a good look at all the defects and triage them, and maybe to add coverity support to Travis.

Closing this bug for now.
